analyzers/src/SonarAnalyzer.CFG/LiveVariableAnalysis/RoslynLiveVariableAnalysis.cs

```diff
@@ -70,10 +72,50 @@ public ISymbol ParameterOrLocalSymbol(IOperation operation)
     protected override State ProcessBlock(BasicBlock block)
     {
         var ret = new RoslynState(this);
-        ret.ProcessBlock(Cfg, block);
+        ret.ProcessBlock(Cfg, block, flowCaptureOperations);
         return ret;
     }
 
+    private void ResolveCaptures()
+    {
+        foreach (var operation in Cfg.Blocks
+                                    .Where(x => x.EnclosingRegion.EnclosingRegionOrSelf(ControlFlowRegionKind.LocalLifetime) is not null)
+                                    .SelectMany(x => x.OperationsAndBranchValue)
+                                    .ToExecutionOrder())
+        {
+            if (IFlowCaptureOperationWrapper.IsInstance(operation.Instance))
+            {
+                var flowCapture = IFlowCaptureOperationWrapper.FromOperation(operation.Instance);
+
+                if (IFlowCaptureReferenceOperationWrapper.IsInstance(flowCapture.Value)
+                    && IFlowCaptureReferenceOperationWrapper.FromOperation(flowCapture.Value) is var captureReference
+                    && flowCaptureOperations.TryGetValue(captureReference.Id, out var capturedOperations))
+                {
+                    AddOrUpdateFlowCaptures(flowCapture.Id, capturedOperations.ToArray());
+                }
+                else
+                {
+                    AddOrUpdateFlowCaptures(flowCapture.Id, flowCapture.Value);
+                }
+            }
+        }
+
+        void AddOrUpdateFlowCaptures(CaptureId id, params IOperation[] operations)
+        {
+            var capturedOperations = flowCaptureOperations.TryGetValue(id, out var existing) ? existing : [];
+            capturedOperations.AddRange(operations);
+            if (flowCaptureOperations.ContainsKey(id))
+            {
+                flowCaptureOperations[id] = capturedOperations;
+            }
+            else
+            {
+                flowCaptureOperations.Add(id, capturedOperations);
+            }
```

Comment:
I didn't give you a good hint on this one. There's better end-to-end working example
```suggestion
            if (!flowCaptureOperations.TryGetValue(id, out var list))
            {
                list = [];
                flowCaptureOperations.Add(id, list);
            }
            list.AddRange(operations);
```

Reply:
I just forgot how references work 🥲